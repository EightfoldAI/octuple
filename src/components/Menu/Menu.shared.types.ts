import { MenuItemTypes, MenuSize, MenuVariant } from './Menu.types';
import { ReactNode } from 'react';

export interface MenuRendererProps {
  items: MenuItemTypes[];
  size?: MenuSize;
  variant?: MenuVariant;
  children?: ReactNode;
  [key: string]: any;
}

export interface MenuRenderFunction {
  (props: MenuRendererProps): JSX.Element;
}
