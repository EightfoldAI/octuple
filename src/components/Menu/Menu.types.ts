import React from 'react';
import { ListProps } from '../List';
import {
  MenuItemButtonProps,
  MenuItemCustomProps,
  MenuItemLinkProps,
  MenuItemSubHeaderProps,
  DropdownMenuItemProps,
} from './MenuItem/MenuItem.types';
import { ButtonProps } from '../Button';

export enum MenuVariant {
  disruptive = 'disruptive',
  primary = 'primary',
  neutral = 'neutral',
}

export enum MenuSize {
  large = 'large',
  medium = 'medium',
  small = 'small',
}

export type MenuItemTypes =
  | MenuItemLinkProps
  | MenuItemButtonProps
  | MenuItemSubHeaderProps
  | MenuItemCustomProps
  | DropdownMenuItemProps;

export interface MenuProps
  extends Omit<
    ListProps<MenuItemTypes>,
    'renderItem' | 'role' | 'itemRole' | 'footer'
  > {
  /**
   * Props for the cancel button
   */
  cancelButtonProps?: ButtonProps;
  /**
   * Props for the ok button
   */
  okButtonProps?: ButtonProps;
  /**
   * Callback when cancel button is clicked
   */
  onCancel?: React.MouseEventHandler<HTMLButtonElement>;
  /**
   * On change callback when menu item is clicked
   * @param value
   */
  onChange?: (value: any) => void;
  /**
   * Callback when ok button is clicked
   */
  onOk?: React.MouseEventHandler<HTMLButtonElement>;
  /**
   * The menu aria role.
   * @default 'menu'
   */
  role?: string;
  /**
   * Size of the menu
   * @default MenuSize.Medium
   */
  size?: MenuSize;
  /**
   * SubHeader of the menu
   * @deprecated 'Unused prop'
   */
  subHeader?: string;
  /**
   * Variant of the menu item
   * @default MenuVariant.neutral
   */
  variant?: MenuVariant;
}

export interface DropdownMenuProps extends MenuProps {
  /**
   * If the dropdown is disabled or not
   */
  disabled?: boolean;
  /**
   * Callback executed on reference element click.
   * @param event
   * @returns (event: React.MouseEvent) => void
   */
  referenceOnClick?: (event: React.MouseEvent) => void;
  /**
   * Callback executed on reference element keydown.
   * @param event
   * @returns (event: React.KeyboardEvent) => void
   */
  referenceOnKeydown?: (event: React.KeyboardEvent) => void;
}
