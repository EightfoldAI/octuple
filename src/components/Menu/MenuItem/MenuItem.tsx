import React, {
  FC,
  ForwardedRef,
  forwardRef,
  JSXElementConstructor,
} from 'react';
import { MenuItemProps, MenuItemType } from './MenuItem.types';
import { MenuItemButton } from './MenuItemButton';
import { MenuItemCustom } from './MenuItemCustom';
import { MenuItemLink } from './MenuItemLink';
import { MenuItemSubHeader } from './MenuItemSubHeader';

const TYPE_TO_MENU_ITEM_MAP: Record<
  MenuItemType,
  JSXElementConstructor<any>
> = {
  [MenuItemType.button]: MenuItemButton,
  [MenuItemType.custom]: MenuItemCustom,
  [MenuItemType.link]: MenuItemLink,
  [MenuItemType.subHeader]: MenuItemSubHeader,
};

export const MenuItem: FC<MenuItemProps> = forwardRef(
  ({ type, ...rest }, ref: ForwardedRef<any>) => {
    const Component = TYPE_TO_MENU_ITEM_MAP?.[type] ?? null;
    return <Component ref={ref} {...rest} />;
  }
);
