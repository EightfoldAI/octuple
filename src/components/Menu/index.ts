export * from './Menu';
export * from './Menu.types';
export * from './MenuItem/MenuItem.types';
