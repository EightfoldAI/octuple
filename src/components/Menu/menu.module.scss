.menu-container {
  background: var(--menu-background-color);
  display: flex;
  flex-direction: column;

  .menu-header-container {
    color: var(--menu-header-color);
    display: flex;
    flex: 1;

    &.large {
      @include octuple-h4();
      padding: $space-ml $space-ml $space-xs;
    }

    &.medium {
      @include octuple-h5();
      padding: $space-m $space-m $space-xs;
    }

    &.small {
      @include octuple-h6();
      padding: $space-s $space-s $space-xs;
    }

    .heading {
      flex: 1;
    }
  }

  .menu-footer-container {
    background: var(--menu-footer-background-color);
    position: sticky;
    bottom: 0;
    border-top: solid 1px var(--menu-border-color);

    &.large {
      padding: $space-l;
    }

    &.medium {
      padding: $space-m;
    }

    &.small {
      padding: $space-s;
    }
  }
}
