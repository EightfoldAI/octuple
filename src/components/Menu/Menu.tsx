'use client';

import React, { FC } from 'react';
import { MenuItemTypes, MenuProps, MenuSize, MenuVariant } from './Menu.types';
import { List } from '../List';
import { MenuItem } from './MenuItem/MenuItem';
import { MenuItemType } from './MenuItem/MenuItem.types';
import { Stack } from '../Stack';
import { Button, ButtonSize, ButtonVariant } from '../Button';
import { useCanvasDirection } from '../../hooks/useCanvasDirection';
import { mergeClasses } from '../../shared/utilities';

import styles from './menu.module.scss';

export const Menu: FC<MenuProps> = ({
  cancelButtonProps,
  classNames,
  header,
  itemClassNames,
  itemProps,
  items,
  itemStyle,
  listType,
  okButtonProps,
  onCancel,
  onChange,
  onOk,
  role = 'menu',
  size = MenuSize.medium,
  style,
  subHeader, // TODO: Remove in v3.0.0
  variant = MenuVariant.neutral,
  ...rest
}) => {
  const htmlDir: string = useCanvasDirection();

  const footerClassNames: string = mergeClasses([
    styles.menuFooterContainer,
    {
      [styles.large]: size === MenuSize.large,
      [styles.medium]: size === MenuSize.medium,
      [styles.small]: size === MenuSize.small,
    },
  ]);

  const headerClassNames: string = mergeClasses([
    styles.menuHeaderContainer,
    {
      [styles.large]: size === MenuSize.large,
      [styles.medium]: size === MenuSize.medium,
      [styles.small]: size === MenuSize.small,
    },
  ]);

  const menuClassNames: string = mergeClasses([
    classNames,
    styles.menuContainer,
    {
      [styles.large]: size === MenuSize.large,
      [styles.medium]: size === MenuSize.medium,
      [styles.small]: size === MenuSize.small,
    },
  ]);

  const menuSizeToButtonSizeMap: Map<MenuSize, ButtonSize> = new Map<
    MenuSize,
    ButtonSize
  >([
    [MenuSize.large, ButtonSize.Large],
    [MenuSize.medium, ButtonSize.Medium],
    [MenuSize.small, ButtonSize.Small],
  ]);

  const getListItem = (item: MenuItemTypes, index: number): React.ReactNode => (
    <MenuItem
      classNames={itemClassNames}
      direction={htmlDir}
      key={`oc-menu-item-${index}`}
      variant={variant}
      size={size}
      onClick={onChange}
      onChange={onChange}
      type={item.type ?? MenuItemType.button}
      {...item}
    />
  );

  const getHeader = (): JSX.Element =>
    header && (
      <div className={headerClassNames}>
        <div className={styles.heading}>{header}</div>
      </div>
    );

  const getFooter = (): JSX.Element =>
    (cancelButtonProps || okButtonProps) && (
      <Stack
        flexGap="s"
        justify="flex-end"
        fullWidth
        classNames={footerClassNames}
      >
        {cancelButtonProps && (
          <Button
            variant={ButtonVariant.Neutral}
            {...cancelButtonProps}
            size={menuSizeToButtonSizeMap.get(size)}
            onClick={onCancel}
          />
        )}
        {okButtonProps && (
          <Button
            variant={ButtonVariant.Primary}
            {...okButtonProps}
            size={menuSizeToButtonSizeMap.get(size)}
            onClick={onOk}
          />
        )}
      </Stack>
    );

  return (
    <List<MenuItemTypes>
      {...rest}
      classNames={menuClassNames}
      footer={getFooter()}
      getItem={getListItem}
      header={getHeader()}
      items={items}
      listType={listType}
      role={role}
      style={style}
    />
  );
};
