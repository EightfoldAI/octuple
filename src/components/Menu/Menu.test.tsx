import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { MenuItemType } from './MenuItem/MenuItem.types';
import { MenuSize, MenuVariant } from './Menu.types';
import { Menu } from './Menu';
import { Dropdown } from '../Dropdown';
import { Button } from '../Button';
import { IconName } from '../Icon';
import { RadioGroup } from '../RadioButton';
import { SelectorSize } from '../CheckBox';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

const MenuOverlay = (args: any) => {
  const menuSizeToSelectorSizeSizeMap = new Map<MenuSize, SelectorSize>([
    [MenuSize.large, SelectorSize.Large],
    [MenuSize.medium, SelectorSize.Medium],
    [MenuSize.small, SelectorSize.Small],
  ]);

  return (
    <Menu
      {...args}
      items={[
        {
          iconProps: {
            path: IconName.mdiCalendar,
          },
          text: 'Date',
          value: 'menu 1',
          counter: '8',
        },
        {
          text: 'Disabled button',
          value: 'menu 2',
          disabled: true,
        },
        {
          type: MenuItemType.subHeader,
          text: 'Sub header',
        },
        {
          type: MenuItemType.link,
          text: 'Twitter link',
          href: 'https://twitter.com',
          target: '_blank',
        },
        {
          type: MenuItemType.custom,
          render: ({ onChange, ...rest }) => (
            <RadioGroup
              {...rest}
              {...{
                ariaLabel: 'Radio Group',
                value: 'Radio1',
                items: [1, 2, 3].map((i) => ({
                  value: `Radio${i}`,
                  label: `Radio${i}`,
                  name: 'group',
                  id: `oea2exk-${i}`,
                })),
                layout: 'vertical',
              }}
              onChange={onChange}
              size={menuSizeToSelectorSizeSizeMap.get(args.size)}
            />
          ),
        },
      ]}
      onChange={(item) => {
        args.onChange(item);
      }}
    />
  );
};

const menuProps: object = {
  variant: MenuVariant.neutral,
  classNames: 'my-menu-class',
  style: {
    color: 'red',
  },
  itemClassNames: 'my-menu-item-class',
  itemStyle: {},
  listType: 'ul',
};

const MenuComponent = (): JSX.Element => {
  return (
    <Dropdown overlay={MenuOverlay(menuProps)}>
      <Button text={'Menu dropdown'} />
    </Dropdown>
  );
};

const LargeMenuComponent = (): JSX.Element => {
  const _menuProps: object = {
    ...menuProps,
    size: MenuSize.large,
  };

  return (
    <Dropdown overlay={MenuOverlay(_menuProps)}>
      <Button text={'Menu dropdown'} />
    </Dropdown>
  );
};

const MediumMenuComponent = (): JSX.Element => {
  const _menuProps: object = {
    ...menuProps,
    size: MenuSize.medium,
  };

  return (
    <Dropdown overlay={MenuOverlay(_menuProps)}>
      <Button text={'Menu dropdown'} />
    </Dropdown>
  );
};

const SmallMenuComponent = (): JSX.Element => {
  const _menuProps: object = {
    ...menuProps,
    size: MenuSize.small,
  };

  return (
    <Dropdown overlay={MenuOverlay(_menuProps)}>
      <Button text={'Menu dropdown'} />
    </Dropdown>
  );
};

describe('Menu', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Should render a menu button', () => {
    const { container } = render(<MenuComponent />);
    const dropdownButton = screen.getByRole('button');
    expect(dropdownButton).toBeTruthy();
    expect(() => container).not.toThrowError();
    expect(container).toMatchSnapshot();
  });

  test('Should show the menu items when the button is clicked', async () => {
    render(<MenuComponent />);
    const dropdownButton = screen.getByRole('button');
    dropdownButton.click();
    await waitFor(() => screen.getByText('Date'));
    const menuitem1 = screen.getByText('Date');
    const menuitem2 = screen.getByText('Disabled button');
    const menuitem3 = screen.getByText('Twitter link');
    const menuitem4 = screen.getByText('Radio1');
    const menuitem5 = screen.getByText('Radio2');
    const menuitem6 = screen.getByText('Radio3');
    expect(menuitem1).toBeTruthy();
    expect(menuitem2).toBeTruthy();
    expect(menuitem3).toBeTruthy();
    expect(menuitem4).toBeTruthy();
    expect(menuitem5).toBeTruthy();
    expect(menuitem6).toBeTruthy();
  });

  test('Should support sub header', async () => {
    render(<MenuComponent />);
    const dropdownButton = screen.getByRole('button');
    dropdownButton.click();
    await waitFor(() => screen.getByText('Sub header'));
    const subHeader = screen.getByText('Sub header');
    expect(subHeader).toBeTruthy();
  });

  test('Should support props such as classNames and style', async () => {
    const { container } = render(<MenuComponent />);
    const dropdownButton = screen.getByRole('button');
    dropdownButton.click();
    await waitFor(() => screen.getByText('Date'));
    expect(container.querySelector('.menu-container')?.classList).toContain(
      'my-menu-class'
    );
    expect(
      (container.querySelector('.menu-container') as HTMLElement).style.color
    ).toContain('red');
  });

  test('Menu is large', async () => {
    const { container } = render(<LargeMenuComponent />);
    const dropdownButton = screen.getByRole('button');
    dropdownButton.click();
    await waitFor(() => screen.getByText('Date'));
    expect(container.querySelector('.menu-container')?.classList).toContain(
      'large'
    );
    expect(container).toMatchSnapshot();
  });

  test('Menu is medium', async () => {
    const { container } = render(<MediumMenuComponent />);
    const dropdownButton = screen.getByRole('button');
    dropdownButton.click();
    await waitFor(() => screen.getByText('Date'));
    expect(container.querySelector('.menu-container')?.classList).toContain(
      'medium'
    );
    expect(container).toMatchSnapshot();
  });

  test('Menu is small', async () => {
    const { container } = render(<SmallMenuComponent />);
    const dropdownButton = screen.getByRole('button');
    dropdownButton.click();
    await waitFor(() => screen.getByText('Date'));
    expect(container.querySelector('.menu-container')?.classList).toContain(
      'small'
    );
    expect(container).toMatchSnapshot();
  });

  test('Menu onChange event is triggered when item is clicked', () => {
    const handleChange = jest.fn();

    const { getByText } = render(
      <Menu
        onChange={handleChange}
        items={[
          {
            iconProps: {
              path: IconName.mdiCalendar,
            },
            text: 'Date',
            value: 'menu 0',
            counter: '8',
            secondaryButtonProps: {
              iconProps: {
                path: IconName.mdiTrashCan,
              },
              onClick: () => {
                console.log('Delete clicked');
              },
            },
          },
        ]}
      />
    );

    fireEvent.click(getByText('Date'));

    expect(handleChange).toHaveBeenCalled();
  });

  test('secondaryButtonProps onClick event is triggered when secondary button is clicked', () => {
    const handleClick = jest.fn();

    const { getByRole } = render(
      <Menu
        items={[
          {
            iconProps: {
              path: IconName.mdiCalendar,
            },
            text: 'Date',
            value: 'menu 0',
            counter: '8',
            secondaryButtonProps: {
              iconProps: {
                path: IconName.mdiTrashCan,
              },
              onClick: handleClick,
            },
          },
        ]}
      />
    );

    fireEvent.click(getByRole('button'));

    expect(handleClick).toHaveBeenCalled();
  });
});
