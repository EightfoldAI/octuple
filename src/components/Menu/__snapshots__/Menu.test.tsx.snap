// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Menu is large 1`] = `
<div>
  <div
    class="main-wrapper"
  >
    <button
      aria-controls="dropdown-"
      aria-disabled="false"
      aria-expanded="true"
      aria-haspopup="true"
      class="button button-default button-medium pill-shape"
      id="dropdown-reference"
      role="button"
      tabindex="0"
    >
      <span
        class="button-text-medium"
      >
        Menu dropdown
      </span>
    </button>
    <div
      class="dropdown-wrapper no-padding open"
      id="dropdown-"
      role="listbox"
      style="position: absolute; top: 4px; left: 0px;"
      tabindex="0"
    >
      <div
        class="my-menu-class menu-container large"
        style="color: red;"
      >
        <ul
          class="list-container vertical"
          role="menu"
        >
          <li
            class="menu-item large neutral my-menu-item-class"
          >
            <button
              class="menu-item-button"
              id="listItem0-"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <span
                aria-hidden="false"
                class="icon-wrapper"
                role="presentation"
              >
                <svg
                  role="presentation"
                  style="width: 24px; height: 24px;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,19H5V8H19M16,1V3H8V1H6V3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3H18V1M17,12H12V17H17V12Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    Date
                  </span>
                  <span>
                    8
                  </span>
                </span>
              </span>
            </button>
          </li>
          <li
            class="menu-item large neutral disabled my-menu-item-class"
          >
            <button
              class="menu-item-button"
              disabled=""
              id="listItem1-"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    Disabled button
                  </span>
                </span>
              </span>
            </button>
          </li>
          <li
            class="menu-item-sub-header large my-menu-item-class"
          >
            <span
              data-disabled="true"
              id="listItem2-"
              tabindex="-1"
              variant="neutral"
            >
              Sub header
            </span>
          </li>
          <li
            class="menu-item large neutral my-menu-item-class"
          >
            <a
              aria-disabled="false"
              class="link-style full-width menu-link"
              href="https://x.com"
              id="listItem3-"
              role="menuitem"
              tabindex="0"
              target="_blank"
            >
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    X link
                  </span>
                </span>
              </span>
            </a>
          </li>
          <li
            class="menu-item-custom large my-menu-item-class"
          >
            <div
              aria-label="Radio Group"
              class="radio-group vertical radio-group-large"
              role="group"
            >
              <div
                class="selector selector-large"
              >
                <input
                  aria-disabled="false"
                  checked=""
                  id="oea2exk-1"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio1"
                />
                <label
                  class=""
                  for="oea2exk-1"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-1-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio1
                  </span>
                </label>
              </div>
              <div
                class="selector selector-large"
              >
                <input
                  aria-disabled="false"
                  id="oea2exk-2"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio2"
                />
                <label
                  class=""
                  for="oea2exk-2"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-2-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio2
                  </span>
                </label>
              </div>
              <div
                class="selector selector-large"
              >
                <input
                  aria-disabled="false"
                  id="oea2exk-3"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio3"
                />
                <label
                  class=""
                  for="oea2exk-3"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-3-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio3
                  </span>
                </label>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu Menu is medium 1`] = `
<div>
  <div
    class="main-wrapper"
  >
    <button
      aria-controls="dropdown-"
      aria-disabled="false"
      aria-expanded="true"
      aria-haspopup="true"
      class="button button-default button-medium pill-shape"
      id="dropdown-reference"
      role="button"
      tabindex="0"
    >
      <span
        class="button-text-medium"
      >
        Menu dropdown
      </span>
    </button>
    <div
      class="dropdown-wrapper no-padding open"
      id="dropdown-"
      role="listbox"
      style="position: absolute; top: 4px; left: 0px;"
      tabindex="0"
    >
      <div
        class="my-menu-class menu-container medium"
        style="color: red;"
      >
        <ul
          class="list-container vertical"
          role="menu"
        >
          <li
            class="menu-item medium neutral my-menu-item-class"
          >
            <button
              class="menu-item-button"
              id="listItem0-"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <span
                aria-hidden="false"
                class="icon-wrapper"
                role="presentation"
              >
                <svg
                  role="presentation"
                  style="width: 20px; height: 20px;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,19H5V8H19M16,1V3H8V1H6V3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3H18V1M17,12H12V17H17V12Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    Date
                  </span>
                  <span>
                    8
                  </span>
                </span>
              </span>
            </button>
          </li>
          <li
            class="menu-item medium neutral disabled my-menu-item-class"
          >
            <button
              class="menu-item-button"
              disabled=""
              id="listItem1-"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    Disabled button
                  </span>
                </span>
              </span>
            </button>
          </li>
          <li
            class="menu-item-sub-header medium my-menu-item-class"
          >
            <span
              data-disabled="true"
              id="listItem2-"
              tabindex="-1"
              variant="neutral"
            >
              Sub header
            </span>
          </li>
          <li
            class="menu-item medium neutral my-menu-item-class"
          >
            <a
              aria-disabled="false"
              class="link-style full-width menu-link"
              href="https://x.com"
              id="listItem3-"
              role="menuitem"
              tabindex="0"
              target="_blank"
            >
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    X link
                  </span>
                </span>
              </span>
            </a>
          </li>
          <li
            class="menu-item-custom medium my-menu-item-class"
          >
            <div
              aria-label="Radio Group"
              class="radio-group vertical radio-group-medium"
              role="group"
            >
              <div
                class="selector selector-medium"
              >
                <input
                  aria-disabled="false"
                  checked=""
                  id="oea2exk-1"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio1"
                />
                <label
                  class=""
                  for="oea2exk-1"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-1-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio1
                  </span>
                </label>
              </div>
              <div
                class="selector selector-medium"
              >
                <input
                  aria-disabled="false"
                  id="oea2exk-2"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio2"
                />
                <label
                  class=""
                  for="oea2exk-2"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-2-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio2
                  </span>
                </label>
              </div>
              <div
                class="selector selector-medium"
              >
                <input
                  aria-disabled="false"
                  id="oea2exk-3"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio3"
                />
                <label
                  class=""
                  for="oea2exk-3"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-3-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio3
                  </span>
                </label>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu Menu is small 1`] = `
<div>
  <div
    class="main-wrapper"
  >
    <button
      aria-controls="dropdown-"
      aria-disabled="false"
      aria-expanded="true"
      aria-haspopup="true"
      class="button button-default button-medium pill-shape"
      id="dropdown-reference"
      role="button"
      tabindex="0"
    >
      <span
        class="button-text-medium"
      >
        Menu dropdown
      </span>
    </button>
    <div
      class="dropdown-wrapper no-padding open"
      id="dropdown-"
      role="listbox"
      style="position: absolute; top: 4px; left: 0px;"
      tabindex="0"
    >
      <div
        class="my-menu-class menu-container small"
        style="color: red;"
      >
        <ul
          class="list-container vertical"
          role="menu"
        >
          <li
            class="menu-item small neutral my-menu-item-class"
          >
            <button
              class="menu-item-button"
              id="listItem0-"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <span
                aria-hidden="false"
                class="icon-wrapper"
                role="presentation"
              >
                <svg
                  role="presentation"
                  style="width: 16px; height: 16px;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19,19H5V8H19M16,1V3H8V1H6V3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3H18V1M17,12H12V17H17V12Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    Date
                  </span>
                  <span>
                    8
                  </span>
                </span>
              </span>
            </button>
          </li>
          <li
            class="menu-item small neutral disabled my-menu-item-class"
          >
            <button
              class="menu-item-button"
              disabled=""
              id="listItem1-"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    Disabled button
                  </span>
                </span>
              </span>
            </button>
          </li>
          <li
            class="menu-item-sub-header small my-menu-item-class"
          >
            <span
              data-disabled="true"
              id="listItem2-"
              tabindex="-1"
              variant="neutral"
            >
              Sub header
            </span>
          </li>
          <li
            class="menu-item small neutral my-menu-item-class"
          >
            <a
              aria-disabled="false"
              class="link-style full-width menu-link"
              href="https://x.com"
              id="listItem3-"
              role="menuitem"
              tabindex="0"
              target="_blank"
            >
              <span
                class="menu-item-wrapper"
              >
                <span
                  class="item-text"
                >
                  <span
                    class="label"
                  >
                    X link
                  </span>
                </span>
              </span>
            </a>
          </li>
          <li
            class="menu-item-custom small my-menu-item-class"
          >
            <div
              aria-label="Radio Group"
              class="radio-group vertical radio-group-small"
              role="group"
            >
              <div
                class="selector selector-small"
              >
                <input
                  aria-disabled="false"
                  checked=""
                  id="oea2exk-1"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio1"
                />
                <label
                  class=""
                  for="oea2exk-1"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-1-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio1
                  </span>
                </label>
              </div>
              <div
                class="selector selector-small"
              >
                <input
                  aria-disabled="false"
                  id="oea2exk-2"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio2"
                />
                <label
                  class=""
                  for="oea2exk-2"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-2-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio2
                  </span>
                </label>
              </div>
              <div
                class="selector selector-small"
              >
                <input
                  aria-disabled="false"
                  id="oea2exk-3"
                  name="group"
                  readonly=""
                  type="radio"
                  value="Radio3"
                />
                <label
                  class=""
                  for="oea2exk-3"
                >
                  <span
                    class="radio-button"
                    id="oea2exk-3-custom-radio"
                  />
                  <span
                    class="selector-label selector-label-end"
                  >
                    Radio3
                  </span>
                </label>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu Should render a menu button 1`] = `
<div>
  <div
    class="main-wrapper"
  >
    <button
      aria-controls="dropdown-"
      aria-disabled="false"
      aria-expanded="false"
      aria-haspopup="true"
      class="button button-default button-medium pill-shape"
      id="dropdown-reference"
      role="button"
      tabindex="0"
    >
      <span
        class="button-text-medium"
      >
        Menu dropdown
      </span>
    </button>
  </div>
</div>
`;
